modules/cas_cache/utils/utils_nvme.c

```diff
@@ -9,6 +9,7 @@
 
 #include <linux/ioctl.h>
 #include <linux/file.h>
+#include <linux/nvme_ioctl.h>
```

Comment:
Looks like this header is not present in older kernels (e.g. 3.19).  We typically handle differences between kernels using the ./configure script. You might want to consider adding a config option (e.g. HAS_NVME_IOCTL_H) which would expand to 1 if nvme_ioctl.h is present and include the file here conditionally, based on #if HAS_NVME_IOCTL_H.